src/main/java/com/kryptnostic/rhizome/hazelcast/processors/AbstractMerger.java

```diff
@@ -0,0 +1,66 @@
+package com.kryptnostic.rhizome.hazelcast.processors;
+
+import java.util.Collection;
+import java.util.Map.Entry;
+
+import com.hazelcast.map.AbstractEntryProcessor;
+
+public abstract class AbstractMerger<K, V extends Collection<T>, T> extends AbstractEntryProcessor<K, V> {
+    private static final long serialVersionUID = 4022386342619821133L;
+
+    private final V           newObjects;
+
+    protected AbstractMerger( V objects ) {
+        this.newObjects = objects;
+    }
+
+    @Override
+    public Object process( Entry<K, V> entry ) {
```

Comment:
I think you could probably do some fancy functional stuff here but this LGTM as is